transmit-lib/src/cmr/transmit/kms.clj

```diff
@@ -121,6 +127,31 @@
                        (pr-str expected-subfield-names)
                        (pr-str subfield-names)))))))
 
+(defn- parse-entries-from-xml
+  "Parses the XML returned by the GCMD KMS. Only when doing the idnnode validation, the kms file
+   is in xml format - concept.xml. 
+   Those lines under conceptBrief tag, with conceptScheme=\"idnnode\", the values of prefLabel are the valid ShortNames.
+   for the DirectoryName. Here is a example:
+   <conceptBrief id=\"296143\" uuid=\"03d77986-98aa-4ad4-9070-2b7a41eadb31\" prefLabel=\"SLOAN\" conceptSchemeId=\"599\" conceptScheme=\"idnnode\" isLeaf=\"true\" status=\"\"/> 
+   Returns a sequence of full hierarchy maps."
+  [keyword-scheme xml-content]
+  (let [parsed-xml (xml/parse (java.io.StringReader. xml-content))
```

Comment:
Depending on what xml-content is, you might be able to use `(xml/parse-str)` to simplify things